prov/gni/src/gnix_msg.c

```diff
@@ -1888,6 +1894,48 @@ static int __smsg_eager_msg_w_data(void *data, void *msg)
 	req = _gnix_match_tag(posted_queue, hdr->msg_tag, 0, FI_PEEK, NULL,
 			      &vc->peer_addr);
 	if (req) {
+		if (req->type == GNIX_FAB_RQ_MRECV) {
+
+			mrecv_req = req;
+			req = _gnix_fr_alloc(ep);
```

Comment:
We need to leave the original ```reg``` returned from ```_gnix_match_tag``` to stay in the posted queue if its not full, so I "save" that pointer by assigning ```mrecv_req``` to the pointer returned from the tag match function.  We also need to update the ```mr_space_left``` and ```mr_buf_addr```.  The new fab request is to track the individual "normal" receive.  Does this make sense?